ethyca/fidesdemo
# 880 update fidesops dep to 1.6.3

## Changes:

* updates fidesops version to 1.6.3
* saas configs now require `type` property
* run pre-install as part of `fidesops-request`
* use dev build of admin ui since prod has type error
* adds new auth scopes
* password for new users needs to be base64 encoded
* adds logging for better visibility when waiting for fidesops to be healthy
* updates mailchimp config/dataset

## Functionality Checklist:

### CI commands
- [x] `make demo`
- [x] `make fidesops-request`
- [x] `make fidesctl-evaluate`

### UI Landing Pages
- [x] demo.html (main demo page)
- [x] Flask Webserver
- [x] Fidesops Admin UI
- [x] Fidesops Privacy Center
- [x] Fidesops API docs
- [x] Fidesctl API docs

### UI Actions
- [x] Connected datastores are visible in the datastore connections space 
- [x] A privacy request can be submitted and denied --> a denial reason should be inputted and then subsequently displayed on the SR detail page) 
- [x] An access request can be submitted and approved --> request should process through the connected datastores and a resulting data package should appear in the connected storage location.
- [x] An erasure request can be submitted and approved --> request should process through the connected datastores and user should be able to confirm that the correct information was erased. 
- [x] New users can be created in the User management space 
- [x] Users can be deleted from the User management space. (Note that you can only delete a user if you are the root user or the user is the user you are logged in as)

## Testing Instructions
1. `docker-compose down` to stop existing docker containers
2. Clear out Docker using `docker rm -f $(docker ps -a -q)` then `docker volume rm -f $(docker volume ls -q)`.
3. Check out this branch in git. Navigate into `fidesops` project, then `git pull` to pull lates changes, then `git checkout 880-update-fidesops-dep`.
4. Install dependencies using `make install`
5. Ensure you can run a demo. See `CI` commands in the `CI commands` section above. `make demo` will get the demo UI up and running to test. **Note - in Admin UI, ensure you're logged in as `fidesopsuser`.
6. If you wish to test mailchimp, you'll have to use `.env` or your own env variables with mailchimp creds, and use `exampleuser@ethyca.com` as the identity for an access request. DO NOT make an erasure request with this identity, since this data has to be manually populated.
